### PR TITLE
Remove downloded bodies from the targets properly

### PR DESCRIPTION
**Pull request by GNUp**
Wrong targets were being removed

### CODE REVIEW

File: sync/src/block/extension.rs [hunk 1]
```diff
@@ -479,6 +479,7 @@ impl Extension {
                 _ => {}
             }
         }
+        // I think it already had to be removed from drain() in line 468
```
Comment (sgkim126) [inline sync/src/block/extension.rs]: No. `remove_target` removes the downloading list also. But `drain` removes the downloaded list only.

Comment (sgkim126) [reply sync/src/block/extension.rs]: By the way, I think this change itself looks good.

Comment (sgkim126) [reply sync/src/block/extension.rs]: Would you cleanup the PR and remove the WIP tag?

Comment (GNUp) [reply sync/src/block/extension.rs]: Okay

**Comment by Kais-DkM** [comment]
This patch is nice, but I didn't experience any problem related to this before.
Could you provide a situation that this patch solves?

File: sync/src/block/extension.rs (hunk 1 above)
Comment (GNUp) [reply sync/src/block/extension.rs]: `exists` were removed from downloading list in `import_bodies()`

**Comment by GNUp** [comment]
@Kais-DkM I just read the code

**Comment by GNUp** [comment]
@Kais-DkM I checked it with integration test

**Pull request merged**